### PR TITLE
Fixed flaky add_texts test

### PR DESCRIPTION
**Pull request by whitead**
* Switched to deterministic embedding - OAI ones keep coming back random
* Removed some stray print statements
* Added a test on embeddings being rank 1 (which was true but I'm paranoid)

### CODE REVIEW

File: tests/test_paperqa.py [hunk 1]
```diff
@@ -779,6 +776,12 @@ def test_sparse_embedding():
         dockey="test",
     )
     assert any(docs.docs["test"].embedding)  # type: ignore[arg-type]
+    assert all(
+        len(np.array(x.embedding).shape) == 1 for x in docs.docs.values()
+    ), "Embeddings should be 1D"
+    assert all(
+        len(np.array(x.embedding).shape) == 1 for x in docs.texts
+    ), "Embeddings should be 1D"
```
Comment (jamesbraza) [inline tests/test_paperqa.py:784]: If you wanna be cheeky:

```python
assert all(
    len(np.array(x.embedding).shape) == 1
    for x in itertools.chain.from_iterable((docs.docs.values(), docs.texts))
), "Embeddings should be 1D"
```

File: tests/test_paperqa.py [hunk 2]
```diff
@@ -1607,27 +1610,46 @@ def disabled_test_memory():
 
 def test_add_texts():
     llm_config = {"temperature": 0.1, "model": "babbage-02"}
-    docs = Docs(llm_model=OpenAILLMModel(config=llm_config))
+    # want to use keyword embedding because
```
Comment (jamesbraza) [inline tests/test_paperqa.py:1613]: Just from my perspective, I am not sure how "keyword embedding" correlates to one wanting to use `SparseEmbeddingModel`

I guess, maybe clarify the comment with info how sparse embedding model is equated with keyword embedding

**Review by jamesbraza — APPROVED** [review]
Nice work

File: tests/test_paperqa.py (hunk 1 above)
Comment (whitead) [reply tests/test_paperqa.py:784]: Beyond me :)

File: tests/test_paperqa.py (hunk 2 above)
Comment (whitead) [reply tests/test_paperqa.py:1613]: yea - typo. Meant to say sparse. Fixed in https://github.com/whitead/paper-qa/commit/9b2eb78a5fb013e2abda279889bff76567999cda

**Pull request merged**